framework/components/dockercompose/billing_platform_service/billing_platform_service.go

```diff
@@ -0,0 +1,248 @@
+	}
+
+	cli, cliErr := client.NewClientWithOpts(
+		client.FromEnv,
+		client.WithAPIVersionNegotiation(),
+	)
+	if cliErr != nil {
+		return nil, errors.Wrap(cliErr, "failed to create docker client")
+	}
+	defer cli.Close()
+
+	// so let's try to connect to a Docker network a couple of times, there must be a race condition in Docker
+	// and even when network sandbox has been created and container is running, this call can still fail
+	// retrying is simpler than trying to figure out how to correctly wait for the network sandbox to be ready
+	networks := []string{framework.DefaultNetworkName}
+	networks = append(networks, in.ExtraDockerNetworks...)
+
+	for _, networkName := range networks {
+		framework.L.Debug().Msgf("Connecting billing-platform-service to %s network", networkName)
+		connectContex, connectCancel := context.WithTimeout(ctx, 30*time.Second)
+		defer connectCancel()
+		if connectErr := connectNetwork(connectContex, 30*time.Second, cli, billingContainer.ID, networkName, identifier); connectErr != nil {
+			return nil, errors.Wrapf(connectErr, "failed to connect billing-platform-service to %s network", networkName)
+		}
+		// verify that the container is connected to framework's network
+		inspected, inspectErr := cli.ContainerInspect(ctx, billingContainer.ID)
+		if inspectErr != nil {
+			return nil, errors.Wrapf(inspectErr, "failed to inspect container %s", billingContainer.ID)
+		}
+
+		_, ok := inspected.NetworkSettings.Networks[networkName]
+		if !ok {
+			return nil, fmt.Errorf("container %s is NOT on network %s", billingContainer.ID, networkName)
+		}
+
+		framework.L.Debug().Msgf("Container %s is connected to network %s", billingContainer.ID, networkName)
+	}
+
+	// get hosts for billing platform service
+	billingExternalHost, billingExternalHostErr := billingContainer.Host(ctx)
+	if billingExternalHostErr != nil {
+		return nil, errors.Wrap(billingExternalHostErr, "failed to get host for Billing Platform Service")
+	}
+
+	// get mapped port for billing platform service
+	billingExternalPort, billingExternalPortErr := findMappedPort(ctx, 20*time.Second, billingContainer, DEFAULT_BILLING_PLATFORM_SERVICE_GRPC_PORT)
+	if billingExternalPortErr != nil {
+		return nil, errors.Wrap(billingExternalPortErr, "failed to get mapped port for Chip Ingress")
+	}
+
+	output := &Output{
+		BillingPlatformService: &BillingPlatformServiceOutput{
+			GRPCInternalURL: fmt.Sprintf("http://%s:%s", DEFAULT_BILLING_PLATFORM_SERVICE_SERVICE_NAME, DEFAULT_BILLING_PLATFORM_SERVICE_GRPC_PORT),
+			GRPCExternalURL: fmt.Sprintf("http://%s:%s", billingExternalHost, billingExternalPort.Port()),
+		},
+	}
+
+	framework.L.Info().Msg("Chip Ingress stack start")
+
+	return output, nil
+}
+
+func composeFilePath(rawFilePath string) (string, error) {
+	// if it's not a URL, return it as is and assume it's a local file
+	if !strings.HasPrefix(rawFilePath, "http") {
+		return rawFilePath, nil
```

Comment:
pls modify it to match what we currently do for chip ingress (I modified it last week, i.e we require that local file is prefixed with `file://`)